### PR TITLE
fix(storage): remove timeout when get compaction group id

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As titled. See https://github.com/singularity-data/risingwave/issues/4427 for detail.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
fix https://github.com/singularity-data/risingwave/issues/4427

### CODE REVIEW

File: src/storage/src/hummock/compaction_group_client.rs [hunk 1]
```diff
@@ -49,7 +49,12 @@ impl CompactionGroupClient for CompactionGroupClientImpl {
         &self,
         table_id: StateTableId,
     ) -> HummockResult<CompactionGroupId> {
-        loop {
+        // The loop executes at most twice.
```
Comment (Li0k) [inline src/storage/src/hummock/compaction_group_client.rs:52]: IMHO, the upper layer (executor) can not handle this Err, this PR just a bug traking ?


Is the panic of the upper layer as expected ？For me , beside using push-based , we need a loop like wait_epoch and some log for traking bug.

Comment (zwang28) [reply src/storage/src/hummock/compaction_group_client.rs:52]: Could just panic instead of return error.

Comment (zwang28) [reply src/storage/src/hummock/compaction_group_client.rs:52]: upper layer **may** handle this Err, as it now knows it's working with an in-existent table, mostly likely deleted.

**Review by Li0k — APPROVED** [review]
Thanks! LGTM

**Pull request merged**